src/jvm/main/org/jetbrains/kotlinx/lincheck/strategy/Strategy.kt

```diff
@@ -116,9 +124,23 @@ abstract class Strategy protected constructor(
  * @return the failure, if detected, null otherwise.
  */
 fun Strategy.runIteration(invocations: Int, verifier: Verifier): LincheckFailure? {
+    var isReplayingSameInvocation = false
     for (invocation in 0 until invocations) {
-        if (!nextInvocation()) return null
-        val result = runInvocation()
+        if (!isReplayingSameInvocation && !nextInvocation()) return null
```

Comment:
Alternatively, we can tweak `nextInvocation()` implementation so it would behave differently after `enableSpinCycleReplay()` was called at the end of previous invocation. 

This way we can keep `runIteration()` implementation simpler, hiding the implementation details of managed strategy.

Reply:
You'll probably only need to modify `verify` method, so it does not actually calls verifier in case of `SpinCycleFoundAndReplayRequired` invocation result.

Reply:
I have hidden the details inside the [ModelChecking/Managed]Startegy